packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -273,16 +293,22 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
     keyringId: string,
   ): Promise<void> {
     this.#assertIsUnlocked();
-    // verify the password and unlock the vault
-    const { toprfEncryptionKey, toprfAuthKeyPair } =
-      await this.#unlockVaultAndGetBackupEncKey();
-
-    // encrypt and store the seed phrase backup
-    await this.#encryptAndStoreSeedPhraseBackup({
-      keyringId,
-      seedPhrase,
-      encKey: toprfEncryptionKey,
-      authKeyPair: toprfAuthKeyPair,
+    await this.#assertPasswordInSync({
+      skipCache: true,
+    });
+    // NOTE don't include #assertPasswordInSync in #withControllerLock since #assertPasswordInSync already acquires the controller lock
```

Comment:
technically, shouldn't this be called within the same lock?
maybe there should be a flag in `assertPasswordInSync` that allows this to be called from within an existing lock.